worldengine/simulations/precipitation.py

```diff
@@ -42,18 +47,61 @@ def _calculate(seed, width, height):
 
         for y in range(height):#TODO: numpy
             y_scaled = float(y) / height
+            n_scale = 1024 / float(height)
             latitude_factor = 1.0 - (abs(y_scaled - 0.5) * 2)
             for x in range(width):
-                n = snoise2(x / freq, y / freq, octaves, base=base)
+                n = snoise2((x * n_scale) / freq, (y * n_scale) / freq, octaves, base=base)
```

Comment:
Would you mind adding some comments? I recently worked into the previous version of the code and found it difficult to understand.